src/makielayout/blocks/legend.jl

```diff
@@ -1,3 +1,48 @@
+function get_n_visible(entry::LegendEntry)
+    n_visible = Ref(0)
+    n_total = Ref(0)
+    foreach_plot_visibility(entry) do vis
+        n_visible[] += Int64(vis[])
+        n_total[] += 1
+    end
+    return n_visible[], n_total[]
+end
+
+function _toggle_over_mouse!(mpos, entry_groups, entryshades)
+    for ((title, entries), shades) in zip(entry_groups, entryshades)
+        for (entry, shade) in zip(entries, shades)
+            # shade and halfshade are Box()es covering the entry
+            bbox = shade.layoutobservables.computedbbox[]
+            if mpos in bbox
+                # determine number of currently visible plot elements
+                n_visible, n_total = get_n_visible(entry)
+                n_total == 0 && return
+                # if not all attached plots have the same state we set all to visible
+                sync_to_visible = n_visible != n_total
+                toggle_visibility!(entry, sync_to_visible)
+                return
+            end
+        end
+    end
+    return
+end
+
+function _set_visibility_ored!(entry_groups)
+    sync_to_visible = false
+    for (_, entries) in entry_groups, e in entries
+        n_visible, n_total = get_n_visible(e)
+        sync_to_visible |= n_visible != n_total
+        sync_to_visible && break
+    end
+    for (_, entries) in entry_groups, e in entries
+        toggle_visibility!(e, sync_to_visible)
+    end
+end
```

Comment:
for naming purposes, this synchronizes visibility of all plots connected to a legend if they are not synchronized and toggles visibility otherwise. So maybe something like `_toggle_synchronized_visibility`?